packages/edge-worker/src/SharedApplicationServer.ts

```diff
@@ -146,7 +146,26 @@ export class SharedApplicationServer {
 		if (this.ngrokUrl) {
 			return this.ngrokUrl;
 		}
-		return process.env.CYRUS_BASE_URL || `http://${this.host}:${this.port}`;
+
+		// If CYRUS_BASE_URL is set, ensure it includes the port
+		if (process.env.CYRUS_BASE_URL) {
+			const baseUrl = process.env.CYRUS_BASE_URL;
+			// Check if the URL already contains a port
+			try {
+				const url = new URL(baseUrl);
+				if (!url.port && this.port !== 80 && this.port !== 443) {
+					// No port specified and not using default HTTP/HTTPS ports
+					url.port = this.port.toString();
+					return url.toString().replace(/\/$/, ""); // Remove trailing slash if present
+				}
+				return baseUrl;
+			} catch {
+				// If URL parsing fails, just return the base URL
+				return baseUrl;
```

Comment:
i think in this case you wouldnt want it as a base case. probably better as a fail case